p2p/peer.go

```diff
@@ -106,6 +215,126 @@ type PeerUpdate struct {
 	Status PeerStatus
 }
 
+// peerStore manages information about peers. It is currently a bare-bones
+// in-memory store of peer addresses, and will be fleshed out later.
+//
+// The main function of peerStore is currently to dispense peers to connect to
+// (via peerStore.Dispense), giving the caller exclusive "ownership" of that
+// peer until the peer is returned (via peerStore.Return). This is used to
+// schedule and synchronize peer dialing and accepting in the Router, e.g.
+// making sure we only have a single connection (in either direction) to peers.
+type peerStore struct {
+	mtx     sync.Mutex
+	peers   map[NodeID]*storePeer
```

Comment:
This name clobbers with the the parent type, perhaps consider `peerState` for individual peer state in a peer store.

Reply:
Yeah, it's a pretty terrible name. Wanted to avoid `peerState` since that's also used in the existing P2P stack in the consensus/mempool reactors. Went with `peerInfo` for now, but this should be improved later.